config/configauth/oidc_authenticator.go

```diff
@@ -116,7 +117,7 @@ func (o *oidcAuthenticator) Authenticate(ctx context.Context, headers map[string
 	return ctx, nil
 }
 
-func (o *oidcAuthenticator) Start(context.Context) error {
```

Comment:
So oidcAuthenticator becomes an extension, right? For it to be available it must be explicitly enabled in the config file, correct?

Reply:
Correct.

Reply:
@jpkrohling what will be the approach for clients? Will this follow the same workflow?

Reply:
No, the client part of the story is probably going to be handled as part of the third issue, #2734. But I remember you had special requirements in this area. Would you mind adding your use case there?

Reply:
Thanks I have added the requirements frim exporters. Not sure if the ticket is the right place but please take a look and advice.

 And if you think clients cannot reuse the authenticators as receivers and the logic is tightly bound to the implementation of clients please let me know. Happy to open an issue accordingly.